drivers/accel/adxl355/adxl355.c

```diff
@@ -0,0 +1,824 @@
+{
+	uint8_t array_raw_x[GET_ADXL355_TRANSF_LEN(ADXL355_XDATA)] = {0};
+	uint8_t array_raw_y[GET_ADXL355_TRANSF_LEN(ADXL355_YDATA)] = {0};
+	uint8_t array_raw_z[GET_ADXL355_TRANSF_LEN(ADXL355_ZDATA)] = {0};
+	int status = 0;
+
+	status = adxl355_read_device_data(dev,ADXL355_ADDR(ADXL355_XDATA),
+					  GET_ADXL355_TRANSF_LEN(ADXL355_XDATA),array_raw_x);
+	if (status != 0)
+		return status;
+	*x = adxl355_accel_conversion(dev, array_raw_x);
+
+	status = adxl355_read_device_data(dev,ADXL355_ADDR(ADXL355_YDATA),
+					  GET_ADXL355_TRANSF_LEN(ADXL355_YDATA),array_raw_y);
+	if (status != 0)
+		return status;
+	*y = adxl355_accel_conversion(dev, array_raw_y);
+
+	status = adxl355_read_device_data(dev,ADXL355_ADDR(ADXL355_ZDATA),
+					  GET_ADXL355_TRANSF_LEN(ADXL355_ZDATA), array_raw_z);
+	if (status != 0)
+		return status;
+	*z = adxl355_accel_conversion(dev, array_raw_z);
+
+	return status;
+}
+
+/***************************************************************************//**
+ * @brief Reads the raw temperature data and converts it to degrees Celsius.
+ *
+ * @param dev      - The device structure.
+ * @param temp     - Temperature output data.
+ *
+ * @return status  - Result of the reading procedure.
+*******************************************************************************/
+int adxl355_get_temp(struct adxl355_dev *dev,
+		     float *temp)
+{
+	uint8_t array_raw_temp[2] = {0, 0};
+	int status = 0;
+
+	status = adxl355_read_device_data(dev, ADXL355_ADDR(ADXL355_TEMP),
+					  GET_ADXL355_TRANSF_LEN(ADXL355_TEMP),array_raw_temp);
+	if (status == 0)
+		*temp = adxl355_temp_conversion(array_raw_temp);
+
+	return status;
+}
+
+/***************************************************************************//**
+ * @brief Reads the status register value.
+ *
+ * @param dev          - The device structure.
+ * @param status_flags - Register value.
+ *
+ * @return status   - Result of the reading procedure.
+*******************************************************************************/
+int adxl355_get_status_register(struct adxl355_dev *dev,
+				union adxl355_status_register_flags *status_flags)
+{
+	int status = 0;
+	uint8_t reg_value = 0;
+
+	status = adxl355_read_device_data(dev, ADXL355_ADDR(ADXL355_STATUS),
+					  GET_ADXL355_TRANSF_LEN(ADXL355_STATUS), &reg_value);
+
+	if (status == 0) {
+		status_flags->value = reg_value;
+	}
+
+	return status;
+}
+
+/***************************************************************************//**
+ * @brief Reads the number of FIFO entries register value.
+ *
+ * @param dev       - The device structure.
+ * @param reg_value - Register value.
+ *
+ * @return status   - Result of the reading procedure.
+*******************************************************************************/
+int adxl355_get_nb_of_fifo_entries(struct adxl355_dev *dev,
+				   uint8_t *reg_value)
+{
+	int status = 0;
+
+	status = adxl355_read_device_data(dev, ADXL355_ADDR(ADXL355_FIFO_ENTRIES),
```

Comment:
there's no point here to add another local variable, return directly. Or even remove this function and use `adxl355_read_device_data` inline.